specifyweb/specify/test_trees.py

```diff
@@ -208,4 +211,238 @@ def test_counts_correctness(self):
             for parent_id, correct in correct_results.items()
         ]
 
+class AddDeleteRanksTest(ApiTests):
+    def setUp(self) -> None:
+        super().setUp()
```

Comment:
if you don't define this method, won't it be inherited from the parent automatically, so this function seems useless? or am I misunderstanding how python works (I am not experienced with it)

Reply:
This is needed when extending an inherited function.  So, the code from the inherited function runs, and then the rest of the code in this function runs.  If we wanted to overwrite the inherited function completely, this line would not be included.  If we just wanted to use the inherited function, then this function would not be needed.

Reply:
hmm but you're not extending the inhereted function. read-up on what these methods do.

Reply:
See this unit test:

https://github.com/specify/specify7/blob/df8a781f9611fb4ab2ffe79cc66b01529b4408bf/specifyweb/stored_queries/tests.py#L91-L109

Reply:
This was just leftover code from refactoring that needed to be removed.